src/azul/plugins/metadata/hca/transform.py

```diff
@@ -977,10 +979,14 @@ def _matrices(self, file: api.File) -> MutableJSON:
             # Stratification values for supplementary files are
             # provided in the 'file_description' field of the file JSON.
             strata_string = file.json['file_description']
-        else:
+            matrix_cell_count = None
+        elif isinstance(file, api.AnalysisFile):
             # Stratification values for analysis files are gathered by
             # visiting the file and using values from the graph.
             strata_string = self._build_strata_string(file)
+            matrix_cell_count = file.matrix_cell_count
+        else:
+            assert False, f'Unexpected file type {type(file)}'
```

Comment:
```suggestion
            assert False, type(file)
```

The rest will be evident from the backtrace and looking at the code. Asserts exist to communicate with a future you (or team member) so you can assume some knowledge of the code base. If this should be requirement rather than an assertion, there should be a human readable message. But I'm unsure wrt require vs assert.